src/Pdf.php

```diff
@@ -125,6 +125,32 @@ public function saveImage($pathToImage)
     }
 
     /**
+     * Save the file as images to the given path.
+     *
+     * @param string $storeDirectory
+     * @param string $fileName     
+     *
+     * @return array $files the full path to the newly created images.
+     */
+    public function extractAllPagesAsImages($storeDirectory, $fileName)
+    {
+        $numberOfPages = $this->getNumberOfPages();
+        if($numberOfPages === 0) return [];
+
+        $files = [];
+        foreach(range(1,$numberOfPages) as $pageNumber) { 
```

Comment:
the foreach loop could be replaced by and `array_map`. In this way the `$files` will be not needed any more, improving readability even more.